share/pvss/pvss.go

```diff
@@ -84,21 +90,33 @@ func EncShares(suite Suite, H kyber.Point, X []kyber.Point, secret kyber.Scalar,
 }
 
 // VerifyEncShare checks that the encrypted share sX satisfies
-// log_{H}(sH) == log_{X}(sX) where sH is the public commitment computed by
+// log_{h}(sH) == log_{X}(sX) where sH is the public commitment computed by
 // evaluating the public commitment polynomial at the encrypted share's index i.
-func VerifyEncShare(suite Suite, H kyber.Point, X kyber.Point, sH kyber.Point, encShare *PubVerShare) error {
+func VerifyEncShare(
+	suite Suite,
+	H kyber.Point,
+	X kyber.Point,
+	sH kyber.Point,
+	encShare *PubVerShare,
+) error {
 	if err := encShare.P.Verify(suite, H, X, sH, encShare.S.V); err != nil {
-		return errorEncVerification
+		return errEncVerification
 	}
 	return nil
 }
 
 // VerifyEncShareBatch provides the same functionality as VerifyEncShare but for
 // slices of encrypted shares. The function returns the valid encrypted shares
 // together with the corresponding public keys.
-func VerifyEncShareBatch(suite Suite, H kyber.Point, X []kyber.Point, sH []kyber.Point, encShares []*PubVerShare) ([]kyber.Point, []*PubVerShare, error) {
+func VerifyEncShareBatch(
+	suite Suite,
+	H kyber.Point,
+	X []kyber.Point,
+	sH []kyber.Point,
+	encShares []*PubVerShare,
+) ([]kyber.Point, []*PubVerShare, error) {
```

Comment:
Wait, what's happening below in case a EncShare doesn't verify?

The error is silently suppressed and its K and E are just skipped?

Is this really what we want? cc @azixus

Reply:
Good point. I agree that just skipping the invalid shares is a bit too silent.

I'm not exactly sure what the approach is though. Maybe we could return a list containing the id of the invalid shares. I'm not quite happy with this as developers may just choose to ignore it and we're back to square one.